src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultArtifactsPath.props

```diff
@@ -20,7 +20,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
   
   <!-- Setting ArtifactsPath automatically opts in to the artifacts output format -->
-  <PropertyGroup Condition="'$(ArtifactsPath)' != ''">
+  <PropertyGroup Condition="'$(ArtifactsPath)' != '' And '$(UsingMicrosoftArtifactsSdk)' != 'true'">
```

Comment:
When a user references Microsoft.Build.Artifacts as a package, its `.props` is imported by `.nuget.g.props` which is imported after this.  So I don't think `UsingMicrosoftArtifactsSdk` will be set yet since this is imported immediately after `Directory.Build.props`.

Reply:
This is only imported early on if `UseArtifactsOutput` or `ArtifactsPath` are set in Directory.Build.props: https://github.com/dotnet/sdk/blob/9d157dca94879daaf6f2d7ea7f8243546bbd2799/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.props#L64-L65

If they are set later on (ie in the project file), then this file will be imported later, during the .targets evaluation: https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultOutputPaths.targets#L42-L43

So if the Microsoft.Build.Artifacts NuGet package is being used and `ArtifactsPath` is set in Directory.Build.props, then the built-in SDK artifacts functionality will be used.  If `ArtifactsPath` is set in the project file (as seems to be suggested in the Microsoft.Build.Artifacts documentation), then the Microsoft.Build.Artifacts logic will be used.

Does this sound OK?  Do you know if anyone using Microsoft.Build.Artifacts is setting `ArtifactsPath` in Directory.Build.props?  IE, something like:

```xml
<Project>
  <PropertyGroup>
    <BaseArtifactsPath>$(MSBuildThisFileDirectory)artifacts</BaseArtifactsPath>
    <ArtifactsPath>$(BaseArtifactsPath)\$(MSBuildProjectName)</ArtifactsPath>
  </PropertyGroup>
</Project>
```

Reply:
@jeffkl I went ahead and merged this because we wanted to get the fix in before the .NET 8 Preview 6 branch,  Let me know what you think about this and we can follow up with another change if need be.